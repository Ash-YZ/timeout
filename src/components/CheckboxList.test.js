import React from 'react';
import { shallow } from 'enzyme';
import CheckBoxList from './CheckboxList'

const changeFn = jest.fn();
const submitFn = jest.fn();

const testProps = {
  title: 'Test Title',
  list: [{ name: 'Person 1' }],
  groupName: 'testGroup',
  handleChange: changeFn,
  handleSubmit: submitFn,
  sumbitLabel: 'Submit'

}
const component = shallow(<CheckBoxList {...testProps} />);

describe('CheckBoxList', () => {
  it('should render correctly ', () => {
    expect(component).toMatchSnapshot()
  })

  it('should trigger changeFn when a checkbox value changes', () => {
    component.find('.cbl-checkbox').first().simulate('change')
    expect(changeFn).toHaveBeenCalled()
  })

  it('should trigger changeFn when checkbox label is clicked', () => {
    component.find('.cbl-label').first().simulate('click')
    expect(changeFn).toHaveBeenCalled()
  })

  it('should trigger submitFn when the submit button is clicked', () => {
    component.find('.cbl-submit').first().simulate('click')
    expect(submitFn).toHaveBeenCalled()
  })
});