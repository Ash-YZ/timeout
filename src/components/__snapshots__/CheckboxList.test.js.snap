// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxList should render correctly  1`] = `
<Fragment>
  <fieldset>
    <legend>
      Test Title
    </legend>
    <div
      key="0"
    >
      <input
        className="cbl-checkbox"
        id="cbl-0"
        name="testGroup"
        onChange={[MockFunction]}
        type="checkbox"
        value="Person 1"
      />
      <label
        className="cbl-label"
        htmlFor="cbl-0"
      >
        Person 1
      </label>
    </div>
  </fieldset>
  <button
    className="cbl-submit"
    onClick={[MockFunction]}
    type="button"
  >
    Submit
  </button>
</Fragment>
`;
